### ISSUE TITLE
Memory Leak in System.IdentityModel.Tokens.Jwt Version 6.12.2

### ISSUE THREAD

**Issue by remunda** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#1724]
Hi,

we're experiencing memory leak issue after upgrading from 6.12.0 -> 6.12.2.
We think its caused by change  #1685 

There are lots of instances of SignatureProvider on `_eventQueue` field in `_signingSignatureProviders`

![image](https://user-images.githubusercontent.com/3290220/137312070-21e5e643-2e99-42af-bd7b-c692651e70a6.png)

![image](https://user-images.githubusercontent.com/3290220/137312450-56e586c0-8172-47fc-a8a2-f70daf38fa96.png)

We're instantiate `SymetricSecurityKey` like this:

```csharp
validationParameters = new TokenValidationParameters
{
    IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(authOptions.Value.JwtTokenSecret!)),
    ValidateIssuer = false,
    ValidateAudience = false,
    ValidateLifetime = false,
    RequireExpirationTime = false,
    RoleClaimType = UserClaims.Role,
    NameClaimType = UserClaims.FullName,
    AuthenticationType = UserClaims.AuthenticationType
};
```

**Comment by brentschmaltz** [comment]
@dannybtsai can you have a look at this please?

@remunda can you provide the details of your application?
what runtime (asp.net, linux, ....)

thanks
brent

**Comment by remunda** [comment]
@brentschmaltz Of course. Its 
- conainer based on docker image mcr.microsoft.com/dotnet/aspnet:5.0 
- so its Debian 10
- running in our kubernetes cluster (limits 2CPU, 2Gi memory)

**Comment by brentschmaltz** [comment]
@remunda i should have asked, when you say lots how many are you seeing?
How does your code use the SignatureProvider?

We cache SignatureProviders based on a KEY+Algorithm+Type, they must be Released after using.
We don't remove them from the Cache when we release, but mark them so that when the cache grows above limits they will be removed and  disposed.

**Comment by brentschmaltz** [comment]
@remunda  we believe this is fixed in 6.14.1.
Please reopen if this is not fixed.

**Comment by dannybtsai** [comment]
@remunda, have you tried the latest version?
